File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.java
class Solution {
    public String getSmallestString(int n, int k) {
        char[] arr = new char[n];
        Arrays.fill(arr, 'a');
        k -= n;
        
        for(int i = n - 1; k > 0; i--)
        {
            arr[i] += Math.min(k, 25);
            k -= 25;
        }
        
        return new String(arr);
    }
}

File: 1663-smallest-string-with-a-given-numeric-value/1663-smallest-string-with-a-given-numeric-value.cpp
class Solution {
public:
    string getSmallestString(int n, int k) {
        string result(n, 'a');
        
        k -= n;
        
        for(int i = n - 1; i >= 0; i--)
        {
            result[i] += (min(k, 25));
            k -= (k >= 25 ? 25 : k);
        }
        
        return result;
    }
};